### ISSUE TITLE
try/catch/break/finally doesn't reset exception stack

### ISSUE THREAD

**Issue by vtjnash** [JuliaLang/julia#31766]
```julia
julia> let
           for i = 1:2
               try
                   throw("No, I am not the cause of your misery")
               catch
                   break
               finally
               end
           end
           error("HI")
       end
```
```
ERROR: HI
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] top-level scope at ./REPL[8]:10
caused by [exception 1]
"No, I am not the cause of your misery"
Stacktrace:
 [1] top-level scope at ./REPL[8]:4
```

**Comment by c42f** [comment]
Odd; lowering is somehow broken for this form, but not for the same thing without the finally block. My tests in `test/exceptions.jl` don't appear to cover the three block syntax.

**Comment by c42f** [comment]
`expand-try` first lowers the three block version to the following nested form which is also broken if passed directly:

```julia
for i=1:1
    try
       try
           error("E1")
       catch
           break
       end
    finally
    end
end
```

**Comment by c42f** [comment]
Should be fixed in #31766.
